dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt

```diff
@@ -322,6 +322,9 @@ internal object CoreFeature {
     private fun shutDownExecutors() {
         uploadExecutorService.shutdownNow()
         persistenceExecutorService.shutdownNow()
+
+        uploadExecutorService.awaitTermination(1, TimeUnit.SECONDS)
```

Comment:
why not using `shutdown` instead of `shutdownNow`. This will wait to finish all the queued tasks before closing the executor and it will not accept any other new task

Reply:
`shutdown` doesn't wait for the task completion as well https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html#shutdown()

Reply:
Oh...the docs are very confusing, first it says that `previously submitted tasks are executed` and then below it say that it doesn't wait for them 🤦 .